catanatron_core/catanatron/models/board.py

```diff
@@ -56,6 +56,12 @@ class Board:
     """
 
     def __init__(self, catan_map=None, initialize=True):
+        self.buildable_subgraph = None
+        
+        # Set must_recompute_buildable_edges to true after an action that potentially changes
+        # the buildable edges
+        self.buildable_edges_cache = {}
+        self.player_port_resources_cache = {}
```

Comment:
Are these new attributes copied over in the `.copy()` method? Is that the reason we need the possible initialization of self.buildable_subgraph in the `buildable_edges` method?